### PR TITLE
STM32U5 Architecture Support 

### PR DESCRIPTION
**Pull request by michi-jung**
## Summary

This is my current state of work on getting NuttX to run on STM32U5. I am especially interested in the TrustZone support of ARMV8-M. The 'nsh' configuration of the b-u585i-iot02a board runs fine as the non-secure companion to STMicroelectronics' port of TrustedFirmware-M.

## Impact

All code is in new subdirectories and should not have any impact on current configurations.

## Testing

I am running a basic 'nsh' configuration fine as a companion to TrustedFirmware-M on B-U585I-IOT02A.

### CODE REVIEW

File: arch/arm/src/stm32u5/Make.defs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+CMN_CSRCS  = arm_assert.c arm_blocktask.c arm_copyfullstate.c arm_createstack.c
+CMN_CSRCS += arm_doirq.c arm_exit.c arm_hardfault.c arm_initialize.c
+CMN_CSRCS += arm_initialstate.c arm_interruptcontext.c arm_mdelay.c arm_memfault.c
+CMN_CSRCS += arm_modifyreg8.c arm_modifyreg16.c arm_modifyreg32.c
+CMN_CSRCS += arm_releasepending.c arm_releasestack.c arm_reprioritizertr.c
+CMN_CSRCS += arm_schedulesigaction.c arm_sigdeliver.c arm_stackframe.c
+CMN_CSRCS += arm_svcall.c arm_systemreset.c arm_trigger_irq.c arm_udelay.c
+CMN_CSRCS += arm_unblocktask.c arm_usestack.c arm_vfork.c arm_switchcontext.c
+CMN_CSRCS += arm_puts.c
```
Comment (no1wudi) [inline arch/arm/src/stm32u5/Make.defs:42]: Please add `arm_tcbinfo.c` to this file.

Comment (michi-jung) [reply arch/arm/src/stm32u5/Make.defs]: @no1wudi Thanks for the comment.  I have added a commit that does this.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by michi-jung** [comment]
@xiaoxiang781216 Thanks for the approval.  I have TrustedFirmware-M running on the secure side and NuttX on the non-secure side.  If you are interested in more details you could have a look at STMicroelectronic's UM2851 document.  Basically, I replaces the 'TFM_Appli_NonSecure.bin' image with the 'Nuttx.bin' image.

**Comment by xiaoxiang781216** [comment]
Ok, Do you have plan to run NuttX on the secure side too?

**Comment by michi-jung** [comment]
No, I do not plan to run NuttX on the secure side.  The ARM Platform Security Architecture recommends to run the RTOS in the Non-secure state and run some less complex software (with a smaller attack surface) providing the trusted services in the secure state.

**Pull request merged**